helpers/osinfo.go

```diff
@@ -242,3 +242,43 @@ func FtraceEnabled() (bool, error) {
 	}
 	return b[0] == '1', nil
 }
+
+type LockdownMode int32
+
+func (l LockdownMode) String() string {
+	return lockdownModeToString[l]
+}
+
+const (
+	NOVALUE LockdownMode = iota
+	NONE
+	INTEGRITY
+	CONFIDENTIALITY
+)
+
+var stringToLockdownMode = map[string]LockdownMode{
+	"none":            NONE,
+	"integrity":       INTEGRITY,
+	"confidentiality": CONFIDENTIALITY,
+}
+
+var lockdownModeToString = map[LockdownMode]string{
+	NONE:            "none",
+	INTEGRITY:       "integrity",
+	CONFIDENTIALITY: "confidentiality",
+}
+
+func Lockdown() (LockdownMode, error) {
+	LockdownFile := "/sys/kernel/security/lockdown"
+	data, _ := os.ReadFile(LockdownFile)
```

Comment:
I think we want to check this error

Reply:
if data is anything else we will default to NOVALUE and error at the end, no ?

Reply:
True, but I think just in case the code changes it would be better to return NOVALUE explicitly here.

Reply:
Since we pass a NOVALUE and an error back, in the call site we also need to check both. Today we just check the error to be not nil and bail out https://github.com/aquasecurity/tracee/blob/a5bc030ec862039d483eb966bff09b16356dd744/cmd/tracee-ebpf/main.go#L87-L90

This breaks when the distro does not have a lockdown file in the location specified.  Alpine 3.15 is an example. Although out of the box it has no lockdown, this check prevents tracee-ebpf from running. At least that's my understanding when trying it out.

Reply:
You're right AND I opened this (in Tracee's side) https://github.com/aquasecurity/tracee/issues/1396. Idea is that we have to bind mount the file through Makefile.tracee-make (so you can execute tracee in the development container) AND, likely, not to fail if the file doesn't exist (like you say so). I have a PR handling the Makefile case, and need to land a fix in libbpfgo as well. I'll try to do that tomorrow.

Reply:
Being addressed at https://github.com/aquasecurity/tracee/pull/1402